File: backend/app/models/news.py
from sqlalchemy import Column, Integer, String, Text, Date, JSON
from app.db.base_class import Base

class News(Base):
    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, index=True)
    date = Column(Date)
    category = Column(String)
    excerpt = Column(Text)
    image = Column(String)
    content = Column(Text, nullable=True)
    slug = Column(String, unique=True, index=True, nullable=True)
    allowed_roles = Column(JSON, default=list)

File: frontend/src/api/news.ts
import api from './axios';

export interface News {
    id: number;
    title: string;
    date: string;
    category: string;
    excerpt: string;
    image: string;
    slug?: string;
    content?: string;
    allowed_roles?: string[];
}

export interface NewsCreate {
    title: string;
    date: string;
    category: string;
    excerpt: string;
    image: string;
    content?: string;
    allowed_roles?: string[];
}

export interface NewsUpdate {
    title?: string;
    date?: string;
    category?: string;
    excerpt?: string;
    image?: string;
    content?: string;
    allowed_roles?: string[];
}

export const getNews = async () => {
    const response = await api.get<News[]>('/news/');
    return response.data;
};

export const getNewsItem = async (id: number) => {
    const response = await api.get<News>(`/news/${id}`);
    return response.data;
};

export const getNewsBySlug = async (slug: string) => {
    const response = await api.get<News>(`/news/slug/${slug}`);
    return response.data;
};

export const createNews = async (news: NewsCreate) => {
    const response = await api.post<News>('/news/', news);
    return response.data;
};

export const updateNews = async (id: number, news: NewsUpdate) => {
    const response = await api.put<News>(`/news/${id}`, news);
    return response.data;
};

export const deleteNews = async (id: number) => {
    await api.delete(`/news/${id}`);
};
